### PR TITLE
feat: remove "AD is watching" state

### LINKED ISSUE #6316: Remove the "AD is Watching" draft-iesg state.
### Describe the issue

Putting a doc into "AD is watching" state makes the AD the action holder. That should IMO not happen, because watching does not imply any action being needed by the AD.

Example: https://datatracker.ietf.org/doc/draft-halpern-gendispatch-antitrust/

### Code of Conduct

- [X] I agree to follow the [IETF's Code of Conduct](https://github.com/ietf-tools/.github/blob/main/CODE_OF_CONDUCT.md)

### LINKED ISSUE #7583: Remove “AD is watching” state
### Description

“AD is watching” has outlived its usefulness, which Robert knows the history of better than I do. It only serves to confuse new ADs now until they learn to ignore it. Let’s get rid of it. 

### Code of Conduct

- [X] I agree to follow the [IETF's Code of Conduct](https://github.com/ietf-tools/.github/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by jennifer-richards**
Migrates any docs in the `watching` state to `idexists`, updating their histories and adding a comment to their event log. Removes all references to the `watching` state from the code. Removes the `create_in_state` choice field from the `edit_info()` view used to begin the IESG process and instead uses the `pub-req` state exclusively.

Fixes #6316
Fixes #7583

### CODE REVIEW

**Comment by jennifer-richards** [comment]
Created as draft because there are still a bunch of references to the `"watching"` state in code and tests that need to be dealt with.

**Comment by jennifer-richards** [comment]
In deciding what to do with docs in the "AD is watching" state (of which there were 60 as of a ~week-old dev DB snapshot), I'm running into #5560 - my first pass at the migration put expired drafts into `dead`, but it is not clear whether that's the right thing to do. From the discussion there, it seems that perhaps `idexists` would be better. Relevant, the first few `watching` drafts I looked at were similar to https://datatracker.ietf.org/doc/draft-murata-kohn-lilley-xml/history/, where the history incorrectly indicates they're in the `dead` state.

**Comment by rjsparks** [comment]
yes, many transitions to dead should now be transitions to idexist. I can't think of any place we should have code choosing the dead state - an AD should have to choose that explicitly.

The code that does is leftover from before we made an idexists state for the draft-iesg state machine

File: ietf/doc/migrations/0024_remove_ad_is_watching_states.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    for doc in Document.objects.filter(states=watching_state):
+        assert doc.type_id == "draft"
+        doc.states.remove(watching_state)
+        doc.states.add(idexists_state)
+        e = StateDocEvent.objects.create(
+            type="changed_state",
+            by=system_person,
+            doc=doc,
+            rev=doc.rev,
+            desc=f"{draft_iesg_state_type.label} changed to <b>{idexists_state.name}</b> from {watching_state.name}",
```
Comment (rjsparks) [inline ietf/doc/migrations/0024_remove_ad_is_watching_states.py:87]: cringing at the html tags, but keeping the set of things to mutate later regular was a good call

File: ietf/templates/doc/draft/change_state.html [hunk 2]
```diff
@@ -12,9 +12,8 @@ <h1>
     </h1>
     {% if state.slug == "dead" %}
         <p class="alert alert-warning my-3">
-           This document is in IESG state "Dead". It is unusual to change
-           this to anything other than "AD is watching", and this should
-           never be used as a replacement for Begin IESG Processing.
+            This document is in IESG state "Dead". It is unusual to change this to
```
Comment (rjsparks) [inline ietf/templates/doc/draft/change_state.html:15]: Add back the "never be used as a replacement for Begin IESG Processing". (New IESG members assume that moving things into pubreq will have all the side-effects that the form/view for Begin IESG Processing applies, but that's not possible.

We could also consider not allowing the document to make any other transition in this particular case.

**Review by rjsparks — CHANGES_REQUESTED** [review]
Once guidance to new ads change requested.

**Review by rjsparks — DISMISSED** [review]

**Review by rjsparks — APPROVED** [review]

**Pull request merged**